server/memphis_handlers_stations.go

```diff
@@ -592,7 +640,13 @@ func (sh StationsHandler) GetAllStationsDetails(shouldGetTags bool) ([]models.Ex
 }
 
 func (sh StationsHandler) GetStations(c *gin.Context) {
-	stations, err := sh.GetStationsDetails()
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("GetStations: Station " + err.Error())
```

Comment:
in some places you return 500, please fix all places to return 401 instead

Reply:
I didn't understand - 500 is a server error and 401 is unauthorized, I don't see where it should be unauthorized